Documentation/test-suites-overview.md

```diff
@@ -0,0 +1,228 @@
+verifying our software at different levels through the different phases of the
+Development Lifecycle.
+
+Due to our microservices architecture, we name our test suites in a different
+way than standards defined by ISTQB (International Software Testing
+Qualifications Board) or other online references. See (ISTQB Foundation Level
+Syllabus)[https://www.istqb.org/downloads/syllabi/foundation-level-syllabus.html].
+
+Nevertheless, we follow QA best practices and principles in verifying our
+software product and it's individual components at different levels.
+
+## Introduction: programming languages
+
+With the exception of the GUI and some client addons in C, the core components
+of Mender are written in Go.
+
+The unit level tests are written i the same language as the test under code
+(usually Go), while for every other test suite we use Python programming
+language and Pytest framework.
+
+## unit tests
+
+Test suite       | unit tests
+---------------- | ----------
+ISTQB equivalent | Component testing
+Test object      | The code inside a single repository
+Source code      | Repository specific, all `*_test.go` files
+How to run       | `go test $(go list ./...)`
+Dependencies     | Some repositories (namely, backend services) require MongoDB server
+
+This is the lowest test level, where the code is closely inspected to exercise
+the code under test in as many circumstances as possible. We collect coverage
+statistics that are published in each repository GitHub page.
+
+## backend acceptance tests
+
+Test suite       | backend acceptance tests
+---------------- | ----------
+ISTQB equivalent | Component integration testing
+Test object      | Interfaces of a single repository
+Source code      | Repository specific, `tests/` directory
+How to run       | See [mendertesting template](https://github.com/mendersoftware/mendertesting/blob/master/.gitlab-ci-check-docker-acceptance.yml) and [integration wrapper](https://github.com/mendersoftware/integration/blob/master/extra/travis-testing/run-test-environment) (example below)
+Dependencies     | Docker, docker-compose
+
+These test suites focus on the interactions and interfaces between integrated
+components or, in our microservices architecture, services.
+
+### Running locally
+
+Acceptance tests are run in a separate container within the same network as the
+service under test an other services when required. Each repository has a Docker
+compose file (usually named `docker-compose-acceptance.yml`) describing other
+services, networks, aliases that are required to run the tests.
+
+This separate container for the tests is built independently for all
+repositories in
+[mender-test-containers](https://github.com/mendersoftware/mender-test-containers/tree/master/backend-acceptance-testing)
+and published to Docker Hub as
+`mendersoftware/mender-test-containers:acceptance-testing`.
+
+To run the test, we use the [run-test-environment wrapper from integration
```

Comment:
```suggestion
To run the tests, we use the [run-test-environment wrapper from integration
```
? Unsure about this one

Reply:
Yes, it should be "tests" or "test suite".